### PR TITLE
fix: use the root of the project as the functions location

### PR DESCRIPTION
**Pull request by NachoVazquez**
## Changes

- Use the configuration root as the location of the Cloudflare Pages functions when using the `directory` mode.

Fixes: #6066 

## Testing

Test the location of the functions folder creation.

## Docs

Docs don't need to be updated, docs already state that the root of the project is the place to generate the functions folder.

### CODE REVIEW

**Comment by NachoVazquez** [comment]
I need help with the tests here. The fixture doesn't seem to work correctly when creating the functions folder, and the previous tests were naive.

**Comment by matthewp** [comment]
I assume the problem is with using relative paths in the test, right? Those are defined here: https://github.com/withastro/astro/blob/main/packages/astro/test/test-utils.js#L168-L171 if it helps with debugging.

**Comment by NachoVazquez** [comment]
Part of the problem is that the functions folder is not being created when I read the elements of the folder in the `console.log`.

File: packages/integrations/cloudflare/src/index.ts [hunk 1]
```diff
@@ -203,7 +203,7 @@ export default function createIntegration(args?: Options): AstroIntegration {
 				}
 
 				if (isModeDirectory) {
-					const functionsUrl = new URL(`file://${process.cwd()}/functions/`);
+					const functionsUrl = new URL(`file://${_config.root}/functions/`);
```
Comment (AirBorne04) [inline packages/integrations/cloudflare/src/index.ts:206]: This is not working for me. `_config.root` is a URL itself.
Changing this line to
`const functionsUrl = new URL("functions", _config.root);`
works for me, also the test is running fine than.

**Review by AirBorne04 — CHANGES_REQUESTED** [review]

File: packages/integrations/cloudflare/src/index.ts (hunk 1 above)
Comment (NachoVazquez) [reply packages/integrations/cloudflare/src/index.ts:206]: Awesome, that helps! Thanks for the feedback! Making the changes now.

Comment (NachoVazquez) [reply packages/integrations/cloudflare/src/index.ts:206]: I made this change, but tests keep failing locally for me. I will push the update so we can have the same reference.

Comment (NachoVazquez) [reply packages/integrations/cloudflare/src/index.ts]: It does seem to work, interesting. 

Thank you :)

File: packages/integrations/cloudflare/test/directory.test.js [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
 			root: './fixtures/basics/',
+			output: 'server',
 			adapter: cloudflare({ mode: 'directory' }),
 		});
+		await fixture.build();
 	});
 
-	it('Builds', async () => {
-		await fixture.build();
+	it('generates the functions folder on under the config root', async () => {
```
Comment (AirBorne04) [inline packages/integrations/cloudflare/test/directory.test.js:18]: can change the text to
`generate functions folder inside the project root`

File: packages/integrations/src/env.d.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+/// <reference types="astro/client" />
```
Comment (AirBorne04) [inline packages/integrations/src/env.d.ts:1]: can you remove this file from the commit?

**Review by AirBorne04 — CHANGES_REQUESTED** [review]
Sorry for being picky 😄

File: packages/integrations/src/env.d.ts (hunk 3 above)
Comment (NachoVazquez) [reply packages/integrations/src/env.d.ts:1]: Sure thing! Sorry for the small details; I'm new to Astro, and I wasn't sure if this was needed.

File: packages/integrations/cloudflare/test/directory.test.js (hunk 2 above)
Comment (NachoVazquez) [reply packages/integrations/cloudflare/test/directory.test.js:18]: That sounds better :)

**Comment by NachoVazquez** [comment]
> Sorry for being picky 😄

Thanks for being picky :)

File: packages/integrations/cloudflare/functions/[[path]].js [hunk 4]
```diff
@@ -0,0 +1,105 @@
+globalThis.process = {
```
Comment (AirBorne04) [inline packages/integrations/cloudflare/functions/[[path]].js:1]: sorry I did miss this one, can you also delete the functions folder, anything that got build should not be added to the repo

Comment (NachoVazquez) [reply packages/integrations/cloudflare/functions/[[path]].js:1]: Good catch!

**Review by AirBorne04 — APPROVED** [review]
Nice job!

**Comment by NachoVazquez** [comment]
> Nice job! Thank you for making Astro better!

Thanks for your help!!!

**Comment by NachoVazquez** [comment]
Hi @AirBorne04 , sorry to bother you, but do you have any info about when the next release of the Cloudflare adapter will happen? I need this changes :) thank you!

**Comment by NachoVazquez** [comment]
Nevermind, I think it is out already! Thank you!

**Pull request merged**